frontend/src/util/search.ts

```diff
@@ -1,11 +1,24 @@
 import { LazyQueryExecFunction, QueryResult } from '@apollo/client'
 import log from '@util/log'
+import moment from 'moment'
 import { useEffect, useRef, useState } from 'react'
 
+import { ClickhouseQuery, QueryInput } from '@/graph/generated/schemas'
+
 export const POLL_INTERVAL = 5000
+const POLL_EXPIRY_MINUTES = 15
 
 // usePollQuery polls a lazy query function and returns a 'more' count
-export function usePollQuery<T, U>({
+export function usePollQuery<
+	T,
+	U extends
+		| {
+				params: QueryInput
+		  }
+		| {
+				query: ClickhouseQuery
```

Comment:
This shouldn't be needed with the new searches (i.e. only `QueryInput` is used), but if its giving a type error maybe we should make a card to clean it up?

Reply:
ahh this is old code before your new changes were merged in, let me see if I can change this